src/bindings/python/src/pyopenvino/graph/ops/loop.cpp

```diff
@@ -0,0 +1,143 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "pyopenvino/graph/ops/loop.hpp"
+
+#include <string>
+
+#include "ngraph/log.hpp"
+#include "openvino/core/node.hpp"
+#include "openvino/op/loop.hpp"
+#include "pyopenvino/graph/ops/util/multisubgraph.hpp"
+
+namespace py = pybind11;
+
+void regclass_graph_op_Loop(py::module m) {
+    py::class_<ov::op::v5::Loop, std::shared_ptr<ov::op::v5::Loop>, ov::Node> cls(m, "loop");
+    cls.doc() = "openvino.impl.op.Loop wraps ov::op::v0::Loop";
+    cls.def(py::init<>());
+    cls.def(py::init<const ov::Output<ov::Node>&, const ov::Output<ov::Node>&>(),
+            py::arg("trip_count"),
+            py::arg("execution_condition"));
+
+    cls.def(
+        py::init([](ov::Node& trip_count, ov::Node& execution_condition) {
+            auto checkNode = [](ov::Node& node) -> bool {
```

Comment:
I suggest to modernize this lambda:
```
            const auto is_constant_or_param = [](ov::Node& node) {
                const auto type_name = std::string{node.get_type_name()};
                return type_name == "Constant" || type_name == "Parameter";
            }
```
You can consider using dynamic_cast to an appropriate type too, this would be the cleanest solution.